core/chains/legacyevm/chain.go

```diff
@@ -245,9 +245,19 @@ func newChain(ctx context.Context, cfg *evmconfig.ChainScoped, nodes []*toml.Nod
 	}
 
 	// note: gas estimator is started as a part of the txm
-	txm, gasEstimator, err := newEvmTxm(opts.DS, cfg.EVM(), opts.AppConfig.EVMRPCEnabled(), opts.AppConfig.Database(), opts.AppConfig.Database().Listener(), client, l, logPoller, opts, headTracker, clientsByChainID)
-	if err != nil {
-		return nil, fmt.Errorf("failed to instantiate EvmTxm for chain with ID %s: %w", chainID.String(), err)
+	var txm txmgr.TxManager
+	var gasEstimator gas.EvmFeeEstimator
+	//nolint:gocritic // ignoring styling issue
```

Comment:
What rule is being suppressed here?

Reply:
It wanted me to covert this if-else to a switch statement which I thought wouldn't be as clear to read

Reply:
Can you reference that in the description?

Reply:
I updated the comment in the latest commit